ReactWindows/ReactNative.Shared/Bridge/ReactBridge.cs

```diff
@@ -108,11 +110,35 @@ public void Dispose()
 
         private void ProcessResponse(JToken response)
         {
+            ProcessResponse(response, false);
+        }
+
+        private void ProcessResponse(JToken response, bool isEndOfBatch)
```

Comment:
>ProcessResponse [](start = 21, length = 15)

I don't trust the `localBatchHasNativeModuleCalls` vs. `_batchHadNativeModuleCalls`. This all gets a lot simpler if you have a single threaded guarantee.